odoo/addons/base/models/res_currency.py

```diff
@@ -323,6 +324,22 @@ def _get_view(self, view_id=None, view_type='form', **options):
                     node[0].set('string', field[1])
         return arch, view
 
+    @api.onchange('active')
```

Comment:
`onchange` is not suitable for this:
- It is only called during changes of the form view, but it should be also called during installation of module (or others modifications).
- `onchange` shouldn't not modify the database for many reasons (by example 'discard' button won't discard change in database which lead your database into an inconsistency state)

Override write/create/unlink instead, it is much more appropriate.

Reply:
I'm +1 for a test, right now I have an hard time understanding what's going on and I'm not even sure this code is right